src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
```

Comment:
If we unmix them it might look something like this?
```php
$providers = [
    'providers' => [
        'google' => [ 
            'name' => 'google',
            'lastseen' => 'some date',
            'lastupdate' => 'some date'
        ],
        'github' => [ 
            'name' => 'github',
            'lastseen' => 'some later date',
            'lastupdate' => 'some later date'
        ],
    ],
    'lastseen' => 'some date',
    'lastupdate' => 'some date'
];
```